source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/imu/imu.py

```diff
@@ -0,0 +1,209 @@
+            env_ids = slice(None)
+        # reset accumulative data buffers
+        self._data.quat_w[env_ids] = 0.0
+        self._data.lin_vel_b[env_ids] = 0.0
+        self._data.ang_vel_b[env_ids] = 0.0
+        self._data.lin_acc_b[env_ids] = 0.0
+        self._data.ang_acc_b[env_ids] = 0.0
+
+    def update(self, dt: float, force_recompute: bool = False):
+        # save timestamp
+        self._dt = dt
+        # execute updating
+        super().update(dt, force_recompute)
+
+    """
+    Implementation.
+    """
+
+    def _initialize_impl(self):
+        """Initializes the sensor handles and internal buffers.
+
+        This function creates handles and registers the provided data types with the replicator registry to
+        be able to access the data from the sensor. It also initializes the internal buffers to store the data.
+
+        Raises:
+            RuntimeError: If the imu prim is not a RigidBodyPrim
+        """
+        # Initialize parent class
+        super()._initialize_impl()
+        # create simulation view
+        self._physics_sim_view = physx.create_simulation_view(self._backend)
+        self._physics_sim_view.set_subspace_roots("/")
+        # check if the prim at path is a rigid prim
+        prim = sim_utils.find_first_matching_prim(self.cfg.prim_path)
+        if prim is None:
+            raise RuntimeError(f"Failed to find a prim at path expression: {self.cfg.prim_path}")
+        # check if it is a RigidBody Prim
+        if prim.HasAPI(UsdPhysics.RigidBodyAPI):
+            self._view = self._physics_sim_view.create_rigid_body_view(self.cfg.prim_path.replace(".*", "*"))
+        else:
+            raise RuntimeError(f"Failed to find a RigidBodyAPI for the prim paths: {self.cfg.prim_path}")
+
+        # Create internal buffers
+        self._initialize_buffers_impl()
+
+    def _update_buffers_impl(self, env_ids: Sequence[int]):
+        """Fills the buffers of the sensor data."""
+        # check if self._dt is set (this is set in the update function)
+        if not hasattr(self, "_dt"):
+            raise RuntimeError(
+                "The update function must be called before the data buffers are accessed the first time."
+            )
+        # obtain the poses of the sensors
+        pos_w, quat_w = self._view.get_transforms()[env_ids].split([3, 4], dim=-1)
+        quat_w = math_utils.convert_quat(quat_w, to="wxyz")
+        # store the poses
+        self._data.pos_w[env_ids] = pos_w + math_utils.quat_rotate(quat_w, self._offset_pos_b)
+        self._data.quat_w[env_ids] = math_utils.quat_mul(quat_w, self._offset_quat_b)
+
+        # obtain the velocities of the sensors
+        lin_vel_w, ang_vel_w = self._view.get_velocities()[env_ids].split([3, 3], dim=-1)
+        # if an offset is present, the linear velocity has to be transformed taking the angular velocity into account
+        lin_vel_w += torch.cross(ang_vel_w, math_utils.quat_rotate(quat_w, self._offset_pos_b), dim=-1)
```

Comment:
`get_velocities` returns the linear velocity at the com of the body rather than the link. We need to transform it back to the link frame using the com offset.
```suggestion
        # obtain the velocities of the link COM
        lin_vel_w, ang_vel_w = self._view.get_velocities()[env_ids].split([3, 3], dim=-1)
        # transform linear velocity from COM to link origin
        com_pos_b, _ = self._view.get_coms()[env_ids].split([3, 4], dim=-1)
        lin_vel_w -= torch.linalg.cross(ang_vel_w, math_utils.quat_rotate(quat_w, com_pos_b),dim=-1)
        # if an offset is present, the linear velocity has to be transformed taking the angular velocity into account
        lin_vel_w += torch.cross(ang_vel_w, math_utils.quat_rotate(quat_w, self._offset_pos_b), dim=-1)
```

Reply:
Makes sense. Does this then fix some of the issues?

Reply:
Yeah this helps fix the linear velocity values. you have so similarly do this on the acceleration.

Reply:
I have a separate branch where I add these in (more efficiently calling com once) and change the unit test to a basic pendulum. 
[imu_com_offset_2](https://github.com/isaac-sim/IsaacLab/tree/feature/imu_com_offset_2)

Reply:
I would say there are still issues with accuracy and precision of the accelerations.

Reply:
that is a great fix already!

Reply:
just trying it out and the lin_vel_b seems to be the same now

Reply:
did you try the ones from the other branch? It also has the acceleration.